docs/languages/csharp/expression-trees.md

```diff
@@ -0,0 +1,132 @@
+﻿# Expression Trees
+Expression Trees are one of the core concepts in computer science language deign.
+In most languages you may have used, expression trees are used by the language
+compiler software as part of the compilation process.
+
+However, many popular languages hide the details of expression trees from the
+developers that use the language.
+
+C# is different.
+
+C# developers can create and interact with Expression Trees.
```

Comment:
I feel like this and the previous sentence are a tad awkward separated like that.  Perhaps something like "C# is different: developers can create and interact with Expression Trees."?